modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/disruptor/inbound/MessagePreProcessor.java

```diff
@@ -268,6 +269,10 @@ private AndesMessage cloneAndesMessageMetadataAndContent(AndesMessage message) {
      */
     private void setMessageID(AndesMessage message) {
         long messageId = idGenerator.getNextId();
+
+        //Tracing message
+        MessageTracer.trace(message, MessageTracer.MESSAGE_ID_MAPPED + " id: " + messageId);
```

Comment:
nope thats not the point. point is we do the concatenation :
MessageTracer.MESSAGE_ID_MAPPED + " id: " + messageId

for each inbound event. which is not something we want to do  unless tracing is enabled.